tools/querytee/proxy.go

```diff
@@ -153,7 +187,15 @@ func NewProxy(cfg ProxyConfig, logger log.Logger, routes []Route, registerer pro
 			preferred = preferredIdx == idx
 		}
 
-		p.backends = append(p.backends, NewProxyBackend(name, u, cfg.BackendReadTimeout, preferred, cfg.BackendSkipTLSVerify))
+		backendCfg := cfg.parsedBackendConfig[name]
```

Comment:
It'd also be good to check that the config file doesn't contain config for backends that haven't been configured - this would help catch typos in the config file.

Reply:
done in https://github.com/grafana/mimir/pull/10081/commits/9533198bf1362dd152128694ad9d7241f70ef31b